erlang:get_stacktrace/0: deprecated in OTP 21

```
$ concuerror --version
Concuerror v0.18-252-gb0933f
$ erl
Erlang/OTP 21 [DEVELOPMENT] [erts-9.3] [source-a61b9c7c6e] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]
$ make pedantic -j
 ERLC  src/concuerror_scheduler.erl
 ERLC  src/concuerror_dependencies.erl
 ERLC  src/concuerror_callback.erl
compile: warnings being treated as errors
src/concuerror_dependencies.erl:50: erlang:get_stacktrace/0: deprecated; use the new try/catch syntax for retrieving the stack backtrace
Makefile:70: recipe for target 'ebin/concuerror_dependencies.beam' failed
make: *** [ebin/concuerror_dependencies.beam] Error 1
make: *** Waiting for unfinished jobs....
compile: warnings being treated as errors
src/concuerror_scheduler.erl:248: erlang:get_stacktrace/0: deprecated; use the new try/catch syntax for retrieving the stack backtrace
Makefile:70: recipe for target 'ebin/concuerror_scheduler.beam' failed
make: *** [ebin/concuerror_scheduler.beam] Error 1
compile: warnings being treated as errors
src/concuerror_callback.erl:1416: erlang:get_stacktrace/0: deprecated; use the new try/catch syntax for retrieving the stack backtrace
src/concuerror_callback.erl:1884: erlang:get_stacktrace/0: deprecated; use the new try/catch syntax for retrieving the stack backtrace
Makefile:70: recipe for target 'ebin/concuerror_callback.beam' failed
make: *** [ebin/concuerror_callback.beam] Error 1
```

--- a/src/concuerror_callback.erl
+++ b/src/concuerror_callback.erl
@@ -1773,7 +1773,7 @@ ets_ops_access_rights_map(Op) ->
     {next          ,_} -> read;
     {select        ,_} -> read;
     {select_delete ,_} -> write;
-    {internal_select_delete ,_} -> write;
+    {internal_select_delete,_} -> write;
     {update_counter,3} -> write;
     {whereis       ,1} -> none
   end.

--- a/src/concuerror_callback.erl
+++ b/src/concuerror_callback.erl
@@ -1413,7 +1413,7 @@ wrapper(InfoIn, Module, Name, Args) ->
               exit  -> Reason
             end,
           exiting(NewReason, Stacktrace, EndInfo);
-        false -> erlang:raise(Class, Reason, erlang:get_stacktrace())
+        false -> erlang:raise(Class, Reason, [])
       end
   end.
 
@@ -1885,7 +1885,7 @@ system_wrapper_loop(Name, Wrapped, Info) ->
             no_reply -> send_message_ack(Report, false, false, false);
             Reason -> ?crash(Reason);
             Class:Reason ->
-              Stacktrace = erlang:get_stacktrace(),
+              Stacktrace = [],
               ?crash({system_wrapper_error, Name, Class, Reason, Stacktrace})
           end;
         {get_info, To} ->

--- a/src/concuerror_callback.erl
+++ b/src/concuerror_callback.erl
@@ -1854,6 +1854,8 @@ system_wrapper_loop(Name, Wrapped, Info) ->
                   receive
                     Msg -> {From, Msg}
                   end;
+                logger ->
+                  throw(no_reply);
                 standard_error ->
                   #concuerror_info{logger = Logger} = Info,
                   {From, Reply, _} = handle_io(Data, {standard_error, Logger}),

--- a/src/concuerror_callback.erl
+++ b/src/concuerror_callback.erl
@@ -1009,6 +1009,7 @@ run_built_in(ets, F, N, [Name|Args], Info)
     ;{F,N} =:= {select_delete, 2}
     ;{F,N} =:= {internal_select_delete, 2}
     ;{F,N} =:= {update_counter, 3}
+    ;{F,N} =:= {whereis, 1}
     ->
   {Tid, System} = check_ets_access_rights(Name, {F,N}, Info),
   case System of
@@ -1773,7 +1774,8 @@ ets_ops_access_rights_map(Op) ->
     {select        ,_} -> read;
     {select_delete ,_} -> write;
     {internal_select_delete ,_} -> write;
-    {update_counter,3} -> write
+    {update_counter,3} -> write;
+    {whereis       ,1} -> none
   end.
 
 %%------------------------------------------------------------------------------

--- a/src/concuerror_dependencies.erl
+++ b/src/concuerror_dependencies.erl
@@ -660,7 +660,8 @@ ets_is_mutating(#builtin_event{mfargs = {_,Op,[_|Rest] = Args}} = Event) ->
     {SelDelete     ,_} when SelDelete =:= select_delete;
                             SelDelete =:= internal_select_delete ->
       from_delete(hd(Rest));
-    {update_counter,3} -> with_key(hd(Rest))
+    {update_counter,3} -> with_key(hd(Rest));
+    {whereis       ,1} -> false
   end.
 
 with_key(Key) ->
@@ -674,6 +675,7 @@ with_key(Key) ->
 ets_reads_keys(Event) ->
   case keys_or_tuples(Event) of
     any -> any;
+    none -> [];
     {matchspec, _MS} -> any; % can't test the matchspec against a single key
     {keys, Keys} -> Keys;
     {tuples, Tuples} ->
@@ -701,7 +703,8 @@ keys_or_tuples(#builtin_event{mfargs = {_,Op,[_|Rest] = Args}}) ->
     {select        ,_} -> {matchspec, hd(Rest)};
     {SelDelete     ,_} when SelDelete =:= select_delete; SelDelete =:= internal_select_delete ->
       {matchspec, hd(Rest)};
-    {update_counter,3} -> {keys, [hd(Rest)]}
+    {update_counter,3} -> {keys, [hd(Rest)]};
+    {whereis       ,1} -> none
   end.
 
 from_insert(undefined, _, _) ->
@@ -713,6 +716,7 @@ from_insert(Table, Insert, InsertNewOrDelete) ->
   fun(Event) ->
       case keys_or_tuples(Event) of
         any -> true;
+        none -> false;
         {keys, Keys} ->
           InsertKeys =
             ordsets:from_list([element(KeyPos, T) || T <- InsertList]),
@@ -742,6 +746,7 @@ from_delete(MatchSpec) ->
   fun (Event) ->
       case keys_or_tuples(Event) of
         any -> true;
+        none -> false;
         {keys, _Keys} -> true;
         {matchspec, _MS} -> true;
         {tuples, Tuples} ->

--- a/src/concuerror_dependencies.erl
+++ b/src/concuerror_dependencies.erl
@@ -45,7 +45,7 @@ dependent(#event{event_info = Info1, special = Special1},
           ?unique(Logger, ?lwarning, Msg, []),
           true;
         {false, _} ->
-          ?crash({undefined_dependency, Info1, Info2, erlang:get_stacktrace()})
+          ?crash({undefined_dependency, Info1, Info2, []})
       end
   end.
 

--- a/src/concuerror_scheduler.erl
+++ b/src/concuerror_scheduler.erl
@@ -243,7 +243,7 @@ explore(State) ->
       get_next_event(State)
     catch
       C:R ->
-        S = erlang:get_stacktrace(),
+        S = [],
         {{crash, C, R, S}, State}
     end,
   case Status of

--- a/tests-real/suites/erlang-litmus/run_litmus.escript
+++ b/tests-real/suites/erlang-litmus/run_litmus.escript
@@ -1,5 +1,5 @@
 #!/usr/bin/env escript
-%%! +S1 -noshell
+%%! -noshell
 
 %%% This script can be used to run Concuerror on specified tests.
 
@@ -41,13 +41,24 @@ main(Tests) ->
    , failed = 0
    , files  = 0
    , finish = false
-   , limit  = 4
+   , limit  = parallelism()
    , tests  = 0
    }).
 
 initialize() ->
-  print_header(),
-  spawn_link(fun() -> loop(#state{}) end).
+  After17 =
+    case erlang:system_info(otp_release) of
+      "R" ++ _ -> false; %% ... 16 or earlier
+      [D,U|_] -> list_to_integer([D,U]) > 17
+    end,
+  case After17 of
+    false ->
+      to_stderr("Skipping because OTP version < 18", []),
+      halt(0);
+    true ->
+      print_header(),
+      spawn_link(fun() -> loop(#state{}) end)
+  end.
 
 loop(#state{done = All, finish = {true, Report}, tests = All} = State) ->
   #state{failed = Failed, files = Files, tests = Tests} = State,
@@ -103,7 +114,7 @@ run_test(File, Test, Server) ->
   Server ! {test, File, Test, Status}.
 
 run_and_get_exit_status(Command) ->
-  Port = open_port({spawn, Command}, [stream, in, eof, hide, exit_status]),
+  Port = open_port({spawn, Command}, [exit_status]),
   get_exit(Port, infinity).
 
 get_exit(Port, Timeout) ->
@@ -139,19 +150,9 @@ extract_tests(File, Server) ->
     true ->
       case compile:file(File, [binary]) of
         error ->
-          After17 =
-            case erlang:system_info(otp_release) of
-              "R" ++ _ -> false; %% ... 16 or earlier
-              [D,U|_] -> list_to_integer([D,U]) > 17
-            end,
-          case After17 of
-            false -> print_test(File, 'n/a', skip);
-            true ->
-              print_test(File, 'compile', error),
-              compile:file(File, [binary, report_errors]),
-              halt(1)
-          end,
-          ok;
+          print_test(File, 'compile', error),
+          compile:file(File, [binary, report_errors]),
+          halt(1);
         {ok, Module, Binary} ->
           {module, Module} = code:load_binary(Module, File, Binary),
           Exports = Module:module_info(exports),
@@ -183,6 +184,7 @@ finish(Server) ->
 %%%-----------------------------------------------------------------------------
 
 print_header() ->
+  to_stderr("Concurrent jobs: ~w", [parallelism()]),
   to_stderr("~-61s~-12s~-7s",["File", "Test", "Result"]),
   print_line().
 
@@ -194,7 +196,6 @@ print_test(File, Test, Status) ->
   Color =
     case Status of
       ok -> "\033[92m";
-      skip -> "\033[94m";
       _ -> "\033[91m"
     end,
   EndC = "\033[0m",
@@ -227,3 +228,6 @@ print_line() ->
 
 to_stderr(Format, Data) ->
   io:format(standard_error, Format ++ "~n", Data).
+
+parallelism() ->
+  erlang:system_info(schedulers).

--- a/tests/runtests.py
+++ b/tests/runtests.py
@@ -188,7 +188,7 @@ if threads == "":
         threads = "1"
 
 # Print header
-print "Concuerror's Testsuite (%d threads)\n" % int(threads)
+print "Concuerror's Testsuite (THREADS=%d)\n" % int(threads)
 print "%-10s %-20s %-50s  %s" % \
       ("Suite", "Module", "(Test,  Bound,  DPOR)", "Result")
 print "---------------------------------------------" + \

--- a/tests/suites/basic_tests/src/ets_new.erl
+++ b/tests/suites/basic_tests/src/ets_new.erl
@@ -9,7 +9,6 @@
 %%%----------------------------------------------------------------------
 %%% Authors     : Ilias Tsitsimpis <iliastsi@hotmail.com>
 %%% Description : Test the `ets:new' instrumentation
-%%%                 (This test should currenty fail)
 %%%----------------------------------------------------------------------
 
 -module(ets_new).

